pkg/provider/akeyless/akeyless.go

```diff
@@ -473,12 +475,12 @@ func (a *akeylessBase) getCertFromSecret(provider *esv1beta1.AkeylessProvider) (
 	}
 
 	ctx := context.Background()
-	res, err := a.secretKeyRef(ctx, &secretRef)
+	cert, err := utils.ResolveSecretKeyRef(ctx, a.kube, a.storeKind, a.namespace, &secretRef)
```

Comment:
The problem with the utils package is that I wouldn't know where to look for a SecretKey resolver. :)

How about, you can put it into the utils folder but create it's own package? Something like `resolvers`?